### PR TITLE
#BN-26 chores: Setup Docker

### PR DESCRIPTION
**Pull request by Honore123**
# PR Description
This PR creates a docker config for the backend whereby it contains both database(Postgres) container
and node js container.
# Description of tasks that were expected to be completed
List all the tasks and sub-tasks that were assigned to you Make sure you check every completed task. list them in this style
- [x] An image of the backend application has been created
- [x] Database is running in a docker container
- [x] Both docker images can communicate while up and running
# Description of tasks that were expected to be completed and that are not yet completed
- [x] None

# How this may be tested?
1. Install docker on your device; visit [https://www.freecodecamp.org/news/the-docker-handbook/#how-to-install-docker](url) to learn more
2. change database name in `.env` to the name in `database/config/config.json` corresponding to the environment that you are working in
3. run `make build`
4. open your browser and visit `localhost:9090` for accessing the node server
5. You can also visit `localhost:8080` to access the Postgres database using admire
6. In adminer choose `PostgreSQL`
7. Server insert `postgres`
8. Fill all the textarea according to `.env`
9. then click login

# Screenshots (If appropriate/applicable)
Screenshot 2023-03-30 at 1 36 44 PM
Node server

Screenshot 2023-03-30 at 1 33 19 PM
Adminer Login page

Screenshot 2023-03-30 at 1 33 47 PM

Adminer dashboard page

# Please check this Checklist before you submit your PR:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My code generates no warnings
- [x] There are no vulnerabilities

### CODE REVIEW

**Review by Lydie260 — DISMISSED** [review]

**Review by kunda4 — DISMISSED** [review]

File: src/index.js [hunk 1]
```diff
@@ -27,15 +28,32 @@ app.use(cors());
...
+const sequelize = new Sequelize(
+  process.env.DATABASE,
+  process.env.DATABASE_USERNAME,
+  process.env.DATABASE_PASSWORD,
+  {
+    host: "postgres",
+    dialect: "postgres",
+  }
+);
+
```
Comment (chrisostome01) [inline src/index.js]: Database initialization has been already initialized. I don't why we have this here

File: src/index.js [hunk 2]
```diff
@@ -1,23 +1,24 @@
...
+import dotenv from "dotenv";
+import cors from "cors";
+import { Sequelize } from "sequelize";
+import i18next from "i18next";
+import Backend from "i18next-fs-backend";
+import i18nextMiddleware from "i18next-http-middleware";
+import swaggerJSDoc from "swagger-jsdoc";
+import swaggerUi from "swagger-ui-express";
+import router from "./routes/routes";
+import options from "./docs/apidoc";
```
Comment (chrisostome01) [inline src/index.js]: let's use prettier eslint to format the codes

**Review by chrisostome01 — CHANGES_REQUESTED** [review]

File: src/database/config/config.json [hunk 3]
```diff
@@ -3,22 +3,22 @@
...
-    "host": "localhost",
+    "host": "postgres",
     "dialect": "postgres"
   },
   "production": {
     "username": "postgres",
     "password": "54321",
     "database": "ecommerce-app-cogito-db",
-    "host": "localhost",
+    "host": "postgres",
```
Comment (chrisostome01) [inline src/database/config/config.json]: Also here I would suggest that we could have different database names for the different use cases.

**Review by chrisostome01 — CHANGES_REQUESTED** [review]

**Review by chrisostome01 — APPROVED** [review]
# LGTM 🚀

**Pull request merged**